pkg/providers/aws/provider_redis.go

```diff
@@ -0,0 +1,280 @@
+		return nil, nil
+	}
+
+	// the cluster doesn't exist, so create it
+	// verify that all values are set or use defaults
+	logrus.Info("creating redis cluster")
+	verifyRedisConfig(redisConfig)
+	input := &elasticache.CreateReplicationGroupInput{
+		AutomaticFailoverEnabled:    aws.Bool(true),
+		Engine:                      aws.String("redis"),
+		ReplicationGroupId:          redisConfig.ReplicationGroupId,
+		CacheNodeType:               redisConfig.CacheNodeType,
+		EngineVersion:               redisConfig.EngineVersion,
+		ReplicationGroupDescription: redisConfig.ReplicationGroupDescription,
+		NumCacheClusters:            redisConfig.NumCacheClusters,
+		SnapshotRetentionLimit:      redisConfig.SnapshotRetentionLimit,
+	}
+	_, err = cacheSvc.CreateReplicationGroup(input)
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// DeleteStorage Delete elasticache replication group
+func (p *AWSRedisProvider) DeleteRedis(ctx context.Context, r *v1alpha1.Redis) error {
+	// resolve redis information for redis created by provider
+	redisConfig, stratCfg, err := p.getRedisConfig(ctx, r)
+	if err != nil {
+		return errorUtil.Wrapf(err, "failed to retrieve aws redis config for instance %s", r.Name)
+	}
+	if redisConfig.ReplicationGroupId == nil {
+		redisConfig.ReplicationGroupId = aws.String(r.Name)
+	}
+
+	// get provider aws creds so the redis cluster can be deleted
+	providerCreds, err := p.CredentialManager.ReconcileProviderCredentials(ctx, r.Namespace)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed to reconcile aws provider credentials")
+	}
+
+	// setup aws redis cluster sdk session
+	cacheSvc := createCacheService(stratCfg, providerCreds)
+
+	// delete the redis cluster
+	return p.deleteRedisCluster(cacheSvc, redisConfig, ctx, r)
+}
+
+func (p *AWSRedisProvider) deleteRedisCluster(cacheSvc elasticacheiface.ElastiCacheAPI, redisConfig *elasticache.CreateReplicationGroupInput, ctx context.Context, r *v1alpha1.Redis) error {
+	// the aws access key can sometimes still not be registered in aws on first try, so loop
+	rgs, err := getReplicationGroups(cacheSvc)
+	if err != nil {
+		return err
+	}
+
+	// check if the cluster has already been deleted
+	var foundCache *elasticache.ReplicationGroup
+	for _, c := range rgs {
+		if *c.ReplicationGroupId == *redisConfig.ReplicationGroupId {
+			foundCache = c
+			break
+		}
+	}
+	// check if replication group does not exist and delete finalizer
+	if foundCache == nil {
+		// remove the finalizer added by the provider
+		resources.RemoveFinalizer(&r.ObjectMeta, DefaultFinalizer)
+		if err := p.Client.Update(ctx, r); err != nil {
+			return errorUtil.Wrapf(err, "failed to update instance as part of finalizer reconcile")
+		}
+		return nil
+	}
+	// check if replication group exists and is available
+	if *foundCache.Status == "available" {
+		// delete the redis cluster that was created by the provider
+		_, err = cacheSvc.DeleteReplicationGroup(&elasticache.DeleteReplicationGroupInput{
```

Comment:
How do we decide whether to do a final snapshot after delete?

Reply:
@maleck13 There is a separate Jira to address final snapshotting during deletion as this ticked ended up being bloated due to OpenShift and AWS being bundled into the same task